enhancements/installer/byo-custom-dns.md

```diff
@@ -0,0 +1,650 @@
+their own LBs in different ways.
+
+Once the Ingress LB is created by the service, its `IP` and/or `Hostname` are
+available in the [`LoadBalancerIngress`](https://github.com/kubernetes/api/blob/cdff1d4efea5d7ddc52c4085f82748c5f3e5cc8e/core/v1/types.go#L4554) field of the `Status` of the LB
+service. The `IP` or the `Hostname` can be used to add the DNS entry for
+*.apps.
+
+For the user's custom DNS solution to be configured with the *.apps entry before
+cluster installation starts, information about the Ingress LB needs to be
+available before the cluster installation starts. So, is it possible for the
+user to pre-create the Ingress LB just like they would the API and API-Int LBs
+and associate this Ingress LB to the LB service started during cluster install?
+
+The following section tries to answer this question per platform.
+
+
+## AWS
+
+AWS does not support the use of the `LoadBalancerIP` so we will have to rely on
+platform specific annotations for an alternative.
+
+1. [`service.beta.kubernetes.io/aws-load-balancer-name`](https://kubernetes-sigs.github.io/aws-load-balancer-controller/v2.2/guide/service/annotations/#load-balancer-name)
+
+This allows us to pre-create an Ingress LB and provide its name as a value for
+this annotation. The user can pre-create the Ingress Lb, provide its name via
+the install-config. Installer can add this value to the [`Ingress Controller
+CRD`](https://github.com/openshift/api/blob/master/operator/v1/0000_50_ingress-operator_00-ingresscontroller.crd.yaml). The cluster-ingress-operator can then add this value to
+the annotation before the LB service is started.
+
+Although the documentation does not specifically mention it, inspecting the
+[`implementation'](https://github.com/rifelpet/aws-alb-ingress-controller/blob/a30c980164db2d9f54596ab971be8d98453c679d/pkg/annotations/constants.go#L52) , it appears that this
+annotation can be used only with NLBs.
+
+2. [`service.beta.kubernetes.io/aws-load-balancer-eip-allocations`](https://kubernetes-sigs.github.io/aws-load-balancer-controller/v2.4/guide/service/annotations/#eip-allocations)
+
+This annotation allows us to provide a list of static IP addresses for an
+external facing LB. This annotation provides the same functionality as
+`LoadBalancerIP` (that AWS does not support) as an annotation. Can only be used
+with NLBs. The static IP per subnet can be used to configure the custom DNS
+solution before installation. It also needs to be provided to OpenShift via
+the install-config. The Installer will make it avilable to the ingress operator
+via the Ingress Controller CRD as in the above case.
+
+## Azure
+
+1. [`LoadBalancerIP`](https://github.com/kubernetes/api/blob/cdff1d4efea5d7ddc52c4085f82748c5f3e5cc8e/core/v1/types.go#L4721)
+
+Providing this input would mean that the Ingress LB would be created by the
+service. But the DNS entry can be configured before cluster installation to
+point to this IP.
```

Comment:
It seems like this statement applies to all three platforms: AWS, Azure, and GCP, right?

Reply:
I don't think AWS supports this field. Simce Azure and GCP do, I have added them in the section for each of these platforms as option 1.